crml/nft/src/lib.rs

```diff
@@ -974,6 +1001,130 @@ decl_module! {
 				None => Err(Error::<T>::NotForFixedPriceSale.into()),
 			}
 		}
+
+		/// Create an offer on a token
+		/// Locks funds until offer is accepted, rejected or cancelled
+		#[weight = 0]
+		fn make_simple_offer (
+			origin,
+			token_id: TokenId,
+			amount: Balance,
+			asset_id: AssetId,
+			marketplace_id: Option<MarketplaceId>,
+		) -> DispatchResult {
+			let origin = ensure_signed(origin)?;
+			ensure!(!amount.is_zero(), Error::<T>::ZeroOffer);
+			ensure!(Self::token_owner((token_id.0, token_id.1), token_id.2) != origin, Error::<T>::IsTokenOwner);
+			let offer_id = Self::next_offer_id();
+			ensure!(offer_id.checked_add(One::one()).is_some(), Error::<T>::NoAvailableIds);
+
+			// ensure the token_id is not currently in an auction
+			if let Some(TokenLockReason::Listed(listing_id)) = Self::token_locks(token_id) {
+				match Self::listings(listing_id) {
+					Some(Listing::<T>::Auction(_)) => return Err(Error::<T>::TokenOnAuction.into()),
+					None | Some(Listing::<T>::FixedPrice(_)) => (),
+				}
+			}
+			// check user has the required funds to make this offer
+			let balance = T::MultiCurrency::free_balance(&origin, asset_id);
```

Comment:
some context:
this is needed to check balance locks since part of free_balance could actually be locked (only applies to CENNZ)

`reserve()` will fail correctly if user has too low balance for every other currency

added mention of this: https://github.com/cennznet/cennznet/issues/578